controller/src/main/java/io/pravega/controller/eventProcessor/impl/EventProcessorCell.java

```diff
@@ -132,11 +143,19 @@ protected final void shutDown() throws Exception {
 
                 // First close the reader, which implicitly notifies reader position to the reader group
                 log.info("Closing reader for {}", objectId);
-                reader.close();
+                try {
+                    reader.close(getCheckpoint());
```

Comment:
I don't understand why we need to do this. In the case we do need to stop the reader at the latest checkpoint, then can we either produce a checkpoint right before calling close or call `readerOffline` as suggested above?